crates/ruff_linter/src/rules/ruff/mod.rs

```diff
@@ -305,6 +305,35 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn range_suppressions_stable() -> Result<()> {
+        let diagnostics = test_path(
+            Path::new("ruff/suppressions.py"),
+            &settings::LinterSettings::for_rules(vec![
+                Rule::UnusedVariable,
+                Rule::AmbiguousVariableName,
+                Rule::UnusedNOQA,
+            ]),
+        )?;
+        assert_diagnostics!(diagnostics);
+        Ok(())
+    }
+
+    #[test]
+    fn range_suppressions_preview() -> Result<()> {
+        let diagnostics = test_path(
+            Path::new("ruff/suppressions.py"),
+            &settings::LinterSettings::for_rules(vec![
+                Rule::UnusedVariable,
+                Rule::AmbiguousVariableName,
+                Rule::UnusedNOQA,
+            ])
+            .with_preview_mode(),
+        )?;
+        assert_diagnostics!(diagnostics);
```

Comment:
This might be a good case for `assert_diagnostics_diff` which snapshots the difference between two settings: E.g. between preview on and off